src/schema/notifications.ts

```diff
@@ -42,17 +93,34 @@ function itemsForKey(notification: any, key: string) {
 function findDataId(
 	client: ZimbraInMemoryCache,
 	partialDataId: string = '$ROOT_QUERY',
-	predicate: (d: string) => any
+	predicate: (d: string) => any,
+	returnFirstResult: Boolean = true
 ) {
 	const data =
 		client && get(client, 'cache.data.data', get(client, 'data.data'));
 	if (!data) {
 		return;
 	}
-	return Object.keys(data).filter(
-		(dataId: string) =>
-			dataId.indexOf(partialDataId) !== -1 && predicate(dataId)
-	)[0];
+	const results = Object.keys(data).filter((dataId: string) => {
+		return dataId.indexOf(partialDataId) !== -1 && predicate(dataId);
+	});
+
+	if (returnFirstResult) {
+		return results[0];
+	}
+
+	return results;
+}
+
+function utcFromDateString(dateString: String) {
+	const transformedDate = `${dateString.slice(0, 4)}-${dateString.slice(
```

Comment:
not sure what is the logic here but better to have something like zm-x-web/src/utils/date/zimbra-date.js